doc/contributing.asciidoc

```diff
@@ -685,6 +688,45 @@ Return:
   - `__magic__` methods
   - other methods
   - overrides of Qt methods
+* Type hinting: the qutebrowser codebase uses type hints liberally to enable
+  static type checking and autocompletion in editors.
+  - We use http://mypy-lang.org/[mypy] in CI jobs to perform static type
+    checking.
+  - Not all of the codebase is covered currently. Complete type hint
+    coverage is most important in areas that heavily use external libraries
+    (like the classes and modules based on Qt) and in areas meant for use
```

Comment:
Can you elaborate on the "most important in areas that heavily use external libraries" part?

Personally, I've just added type hints to *all* new code, unless I forget doing so. I'd actually suggest that instead - we don't (yet?) enforce type hints for all new code, but I'd heavily suggest it in the contribution docs. Thoughts?

Reply:
I guess I'm less thinking of writing new code but modifying existing code in the codebase which doesn't have type hints. Expecting non-experts to add type hints to existing code they are modifying is generally expecting a bit much I think. So I wanted to offer some guidance based on what seems to be existing conventions (based on the mypy per modules configs and my memory, but I didn't actually survey what files have good coverage or anything). Also I feel like most python code I write mostly uses primitives where they convey less information. But I haven't fact checked that assumption anyway. Also maybe I'm doing it wrong.

Anyway, I want new contributors to be able to understand the current state and what expectations are for them. I'm not wedded to this instruction if it's not helpful. Probably I will drop this bit and merge this bullet with the next one (not all the codebase is covered, because gradual typing, please add them where you can).